tools/failures.py

```diff
@@ -244,22 +267,32 @@ def parse_args(argv=None):
     options = parser.parse_args(argv)
     return options
 
+def analyzeFailures(startDate, endDate, testmode, quick=False):
+    failures = getRawData(startDate, endDate)
+    print "date: %s, failures: %s" % (endDate, len(failures))
+    target = 100 # 100% detection
+
+    if quick:
+        to_remove, total_detected = sanity_check(failures, target, endDate)
+        if len(to_remove) > 0:
+            print "no need for a full run on date: %s, %s, %s" % (endDate, len(to_remove), total_detected)
+            communicate(failures, to_remove, total_detected, testmode, endDate, results=False)
+            return
+
+    # no quick option, or quick failed due to new failures detected
+    print "-- need to do a full run on %s" % (endDate)
+    to_remove, total_detected = seta.depth_first(failures, target)
+    communicate(failures, to_remove, total_detected, testmode, endDate)
+
 if __name__ == "__main__":
     options = parse_args()
 
     if options.start_date and options.end_date:
         print_diff(options.start_date, options.end_date)
     else:
-        failures = getRawData()
-        targets = [100.0]
-        for target in targets:
-            if options.quick:
-                to_remove, total_detected = sanity_check(failures, target)
-                if len(to_remove) > 0:
-                    communicate(failures, to_remove, total_detected, options.testmode, results=False)
-                    continue
-
-            # no quick option, or quick failed due to new failures detected
-            to_remove, total_detected = seta.depth_first(failures, target)
-            communicate(failures, to_remove, total_detected, options.testmode)
+        startDate = datetime.datetime(2014,8,13)
+        endDate = datetime.datetime(2014,11,13) # 3 months minimum
```

Comment:
Maybe use options.start_date and options.end_date for this? I'm not sure what print_diff is doing above, but it doesn't seem that useful.